src/v0/parity.rs

```diff
@@ -33,6 +33,65 @@ pub struct DoughnutV0 {
     pub signature: H512,
 }
 
+impl DoughnutV0 {
+    fn encode_with_signature_optional(&self, encode_signature: bool) -> Vec<u8> {
```

Comment:
I don't like how close the naming of this function is to:
`encode_to_with_signature_optional`

Also, this function is only called once with its encode_signature always set to false, I suggest removing it and putting the vector object in the calling `payload` function.